src/lib/y2storage/autoinst_profile/drive_section.rb

```diff
@@ -572,13 +572,18 @@ def use_value_from_string(use)
 
       # Determine whether snapshots are enabled
       #
-      # Currently AutoYaST does not support enabling/disabling snapshots
-      # for a partition but for the whole disk/volume group.
+      # Currently AutoYaST only supports enabling/disabling snapshots
+      # for the root filesystem and this setting is specified at
+      # drive section level.
       #
       # @param filesystems [Array<Y2Storage::Filesystem>] Filesystems to evaluate
-      # @return [Boolean] true if snapshots are enabled
+      # @return [Boolean,nil] true if snapshots are enabled; false if they are not enabled;
+      #   nil if the root filesystem is not applicable.
       def enabled_snapshots?(filesystems)
-        filesystems.any? { |f| f.respond_to?(:snapshots?) && f.snapshots? }
+        root_fs = filesystems.find { |f| f.respond_to?(:root?) && f.root? }
```

Comment:
As discussed offline, checking if the filesystem responds to `:root?` can be omitted.

Reply:
Actually, I found a potential problem when calling `enabled_snapshots?` from `#init_from_disk` after removing the condition.